lms/djangoapps/teams/views.py

```diff
@@ -541,6 +562,15 @@ def get_queryset(self):
         """Returns the queryset used to access the given team."""
         return CourseTeam.objects.all()
 
+    def delete(self, request, team_id):
+        """DELETE /api/team/v0/teams/{team_id}"""
+        team = get_object_or_404(CourseTeam, team_id=team_id)
+        self.check_object_permissions(request, team)
+        # Note: also deletes all team memberships associated with this team
+        log.info('user %d deleted team %s', request.user.id, team_id)
+        team.delete()
```

Comment:
I don't think there are any exceptions that need to be covered, actually. Django will throw an error if an object has its `delete` method called multiple times, but a separate request will get a separate object, so no error will be thrown. MySQL doesn't care if you try to delete nonexistent data.